projects/sienna/index.js

```diff
@@ -0,0 +1,152 @@
+                    limit: 10,
+                    start: start
+                }
+            }
+        });
+        if (result && result.entries && result.entries.length) {
+            markets = markets.concat(result.entries);
+            start = markets.length;
+        } else grabMarkets = false;
+    }
+
+    return markets;
+}
+
+async function Lend() {
+    const markets = await getLendMarkets();
+    const block = await queryClient.getHeight();
+    return Promise.all(markets.map(async (market) => {
+        const marketState = await queryClient.queryContractSmart(market.contract.address, {
+            state: {
+                block
+            }
+        });
+        const exchange_rate = await queryClient.queryContractSmart(market.contract.address, {
+            exchange_rate: {
+                block
+            }
+        });
+        const underlying_asset = await queryClient.queryContractSmart(market.contract.address, { underlying_asset: {} });
+        const token = await TokenInfo(underlying_asset.address);
+        return {
+            symbol: token.symbol,
+            tokens_supplied: new BigNumber(marketState.total_supply).times(exchange_rate).div(new BigNumber(10).pow(token.decimals)).toNumber(),
+            tokens_borrowed: new BigNumber(marketState.total_borrows).div(new BigNumber(10).pow(token.decimals).toNumber()).toNumber(),
+        };
+    }));
+}
+
+async function StakedTokens() {
+    const siennaToken = await TokenInfo(SIENNA_TOKEN_ADDRESS);
+    const stakedTokens = await Promise.all(SIENNA_SINGLE_SIDED_POOLS.map(async (pool) => {
+        let total_locked;
+        if (pool.version === 3) {
+            const fetchedPool = await queryClient.queryContractSmart(pool.address, { rewards: { pool_info: { at: new Date().getTime() } } });
+            total_locked = fetchedPool.rewards.pool_info.staked;
+        } else {
+            const fetchedPool = await queryClient.queryContractSmart(pool.address, { pool_info: { at: new Date().getTime() } });
+            total_locked = fetchedPool.pool_info.pool_locked;
+        }
+        return new BigNumber(total_locked).div(new BigNumber(10).pow(siennaToken.decimals)).toNumber();
+    }));
+    return stakedTokens.reduce((total, value) => total + value, 0);
+}
+
+async function TVL() {
+    const balances = {};
+
+    const pairs_volumes = await PairsVolumes();
+    await Promise.all(pairs_volumes.map(async volume => {
+        if (utils.symbolsMap[volume.symbol]) await sdk.util.sumSingleBalance(balances, utils.symbolsMap[volume.symbol], volume.tokens);
+    }));
+
+    const staked_tokens = await StakedTokens();
+    if (staked_tokens) await sdk.util.sumSingleBalance(balances, "sienna", staked_tokens);
+
+    const lend_data = await Lend();
+
+    await Promise.all(lend_data.map(async volume => {
+        if (utils.symbolsMap[volume.symbol]) await sdk.util.sumSingleBalance(balances, utils.symbolsMap[volume.symbol], volume.tokens_supplied);
```

Comment:
can this just count liquidity (supply - borrow)? Then the borrowed amount can be exported as a separate 'borrowed' key - this lets defillama users decide whether or not they want stats to include borrowed funds

Reply:
You are right, fixed the lend markets liquidity by subtracting the borrowed amount of tokens from the supplied amount.